flight/PiOS/Common/pios_mpu.c

```diff
@@ -465,6 +520,24 @@ static int32_t PIOS_MPU_Common_Init(void)
 	mpu_dev->accel_range = PIOS_MPU_SCALE_8G;
 	mpu_dev->gyro_range = PIOS_MPU_SCALE_1000_DEG;
 
+	int ret = PIOS_SENSORS_RegisterCallback(PIOS_SENSOR_GYRO,
+			PIOS_MPU_callback_gyro, mpu_dev);
+
+	PIOS_Assert(!ret);
```

Comment:
Using asserts in here rather than returning an error means the user doesn't get a LED flash code to indicate where the problem happened so we get less to go on in bug reports..